validator/client/grpc-api/grpc_node_client.go

```diff
@@ -29,8 +30,13 @@ func (c *grpcNodeClient) ListPeers(ctx context.Context, in *empty.Empty) (*ethpb
 	return c.nodeClient.ListPeers(ctx, in)
 }
 
-func (c *grpcNodeClient) IsHealthy(context.Context) bool {
-	panic("function not supported for gRPC client")
+func (c *grpcNodeClient) IsHealthy(ctx context.Context) bool {
+	_, err := c.nodeClient.GetHealth(ctx, &ethpb.HealthRequest{})
+	if err != nil {
+		log.WithError(err).Error("failed to get health of node")
+		return false
+	}
+	return true
```

Comment:
Why does this always return true if any response was received? Seems like this method would identify any node as "healthy" when it returned a non-200 response code.

Reply:
that's all this endpoint does

 // Retrieve the current health of the node.
    rpc GetHealth(HealthRequest) returns (google.protobuf.Empty) {
        option (google.api.http) = {
            get: "/eth/v1alpha1/node/health"
        };
    }

Reply:
on grpc so it should be fine.